test/localstorage.test.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * @vitest-environment jsdom
```

Comment:
I think it's better to create `Window` ourselves. Otherwise we depend on test framework implementation

We do this with happy-dom

Reply:
I'm still trying to figure out what is causing this. It is only happening with jsdom and not happy-dom.
I'll probably not use localStorage at all when I figure out the cause.

Reply:
> I'm still trying to figure out what is causing this. It is only happening with jsdom and not happy-dom. I'll probably not use localStorage at all when I figure out the cause.

This aproach should help narrow it down, since it is still possible the error might be in Vitest

Reply:
https://github.com/jsdom/jsdom/issues/2318 - you cannot actually call defineProperty to change implementation on a localStorage.

You need to replace entire localStorage implementation if you need to mock it: https://github.com/facebook/jest/issues/6798#issuecomment-437537691

Reply:
`jest.spyOn(window.localStorage.__proto__, 'setItem');` should work without any additional imports.

Hehe

Reply:
Yeah, maybe just need to document it somewhere